controllers/rate_limiting_wasmplugin_controller.go

```diff
@@ -267,22 +271,6 @@ func (r *RateLimitingWASMPluginReconciler) wasmRateLimitPolicy(ctx context.Conte
 	routeWithEffectiveHostnames := route.DeepCopy()
 	routeWithEffectiveHostnames.Spec.Hostnames = hostnames
 
-	// Policy limits may be overridden by a gateway policy for route policies
-	if kuadrantgatewayapi.IsTargetRefHTTPRoute(rlp.GetTargetRef()) {
-		filteredPolicies := utils.Filter(affectedPolices, func(p kuadrantgatewayapi.Policy) bool {
-			return kuadrantgatewayapi.IsTargetRefGateway(p.GetTargetRef()) && p.GetUID() != rlp.GetUID()
-		})
-
-		for _, policy := range filteredPolicies {
-			p := policy.(*kuadrantv1beta2.RateLimitPolicy)
-			if p.Spec.Overrides != nil {
-				rlp.Spec.CommonSpec().Limits = p.Spec.Overrides.Limits
```

Comment:
Here lies the bug. We are adding the overridden policy to the config, only with the limit definitions replaced with the ones from the override. Except that this will never find a match in the Limitador CR.

Reply:
Hmmm. A minute has passed and I'm already having second doubts about this fix...

Maybe what we're doing in the WasmPlugin is not wrong after all, but rather what we do in the Limitador CR is?

WDYT @eguzki @KevFan

Reply:
The difference is basically if we want a single counter for all routes or one counter per route.

Reply:
I've given this another iteration after speaking with @eguzki.

We cannot take the simpler approach of falling back to the override limit definitions, otherwise we'd be sharing the same counters between all overridden routes. Instead, we really want to duplicate limit definitions for each independent `domain`/`namespace`, even if the configuration of the counters end up being identical.

With that, the actual fix became:
1. keeping the wasm config unchanged – only minor refactoring now in this PR;
2. duplicating the entries in the limits config (Limitador CR) for each gateway for an overridden policy that affects multiple gateways

(2) required changing the hash that ensures unique limit names (conditions), both in the wasm config and Limitador CR. Another option could be adding the name of the gateway again to domain/namespace (as it once was the case long ago.)